types/cross-spawn/index.d.ts

```diff
@@ -1,16 +1,36 @@
 // Type definitions for cross-spawn 6.0
 // Project: https://github.com/moxystudio/node-cross-spawn
 // Definitions by: Alorel <https://github.com/Alorel>
+//                 ExE Boss <https://github.com/ExE-Boss>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 /// <reference types="node" />
 
 import * as child_process from 'child_process';
 
-declare function spawn(command: string, args?: string[], options?: child_process.SpawnOptions): child_process.ChildProcess;
-
 declare namespace spawn {
-    const sync: typeof child_process.spawnSync;
+	/**
+	 * The `spawn()` method spawns a new process using the given `command`, with
+	 * command line arguments in `args`. If omitted, `args` defaults to an empty array.
+	 */
+	const spawn: typeof child_process.spawn;
+
+	/**
+	 * The `spawn.sync()` method spawns a new process using the given `command`, with
+	 * command line arguments in `args`. If omitted, `args` defaults to an empty array.
+	 */
+	const sync: typeof child_process.spawnSync;
 }
 
+/**
+ * The `spawn()` method spawns a new process using the given `command`, with
+ * command line arguments in `args`. If omitted, `args` defaults to an empty array.
+ */
+declare function spawn(command: string, options: child_process.SpawnOptions): child_process.ChildProcess;
```

Comment:
Why define a type for this at all if you're using `typeof child_process.spawn`?
Alternatively, either define a type for spawn everywhere, or just use the type from child_process - I don't think it makes sense to do both.

Reply:
The issue here is that, using `typeof child_process.spawn` here causes incompatibilities with `namespace spawn`.

This is the only way to use both `namespace` and `function`.